src/main/java/io/vertx/mqtt/impl/MqttClientImpl.java

```diff
@@ -640,6 +675,14 @@ void handleUnsuback(int unsubackMessageId) {
   void handlePuback(int pubackMessageId) {
 
     synchronized (this.connection) {
+      io.netty.handler.codec.mqtt.MqttMessage removedPacket = qos1outbound.remove(pubackMessageId);
+
+      if (removedPacket == null) {
+        return;
```

Comment:
Add a log warning here, because a PUBACK is arrived but we don't have the corresponding message-id in the qos1outbound queue ... it's a weird situation :-)